### PR TITLE
fix: buy now switch in searchbar

### LINKED ISSUE #3372: "Buy Now" Button not work on collection page (BSX)
### What happened?

BUY NOW filtering does not seem to work at the level of individual collections

### Please reproduce in steps

Info in the attached video

### Expected Behavior

I've expected ...

### What browsers are you seeing the problem on?

Firefox, Chrome

### Are you logged in?

Yes

### At which address did you encounter bug?

kodadot.xyz

### Screenshots

https://user-images.githubusercontent.com/81104034/177830628-03ae1b25-fa8b-46ec-af09-1a79f036018e.mp4

### Relevant log output

_No response_

### Payment link for reward

https://kodadot.xyz/transfer?target=FqCJeGcPidYSsvvmT17fHVaYdE2nXMYgPsBn3CP9gugvZR5&usdamount=0&donation=true

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kkukelka**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

https://user-images.githubusercontent.com/17953978/178101881-d75753fa-7f3b-4d93-8844-c90f63580a19.mov

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3372
- [ ] Requires deployment <rubick/magick_issue>
- [x] Fixed how searchQuery params were initialised and the way changes were emitted from searchbar

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've tested it at </bsx/explore?page=1&listed=true>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://kodadot.xyz/transfer/?target=EqdyzrzVmeHwMdMwvPeCMnNdbuQDbD3YrjY93xq9Ln3jUGW)

### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
✅ Works well on collections page

Need to be fixed on:
- `/bsx/explore` Gallery Tab
- `/bsx/collection/{id}` collection details (tested on /bsx/collection/4011142362)

**Comment by kkukelka** [comment]
@roiLeo
I've fixed the `/bsx/collection/{id}` (e.g.`/bsx/collection/2526524581`), since it also uses the `SearchBarCollection.vue` component.
https://user-images.githubusercontent.com/17953978/178134307-0a8704d4-f03e-408f-a73d-b28e6a66b904.mov

But the gallery one is probably out of scope for this ticket, as it uses an entire different Search component (`SearchBar.vue`) and none of the search functionality seems to work for Basilisk
-> I have created a separate Issue for that one #3398

**Comment by petersopko** [comment]
>  I have created a separate Issue for that one #3398

tracking this here
- https://github.com/kodadot/nft-gallery/issues/3293

File: components/rmrk/Gallery/CollectionItem.vue [hunk 1]
```diff
@@ -303,9 +301,14 @@ export default class CollectionItem extends mixins(
...
-        price: { greaterThan: '0' },
-      })
+      const prefix = this.$store.getters.currentUrlPrefix
+      if (prefix === 'bsx') {
+        params.push({ price_gt: '0' })
+      } else {
+        params.push({
+          price: { greaterThan: '0' },
+        })
+      }
```
Comment (roiLeo) [inline components/rmrk/Gallery/CollectionItem.vue:311]: you can use `this.urlPrefix` here instead

& I would change instead 
```ts
if (this.urlPrefix === 'rmrk') {
  params.push({
    price: { greaterThan: '0' },
  })
} else {
  params.push({ price_gt: '0' })
}
```
so we make it work for monsama too

File: components/rmrk/Gallery/Search/SearchBarCollection.vue [hunk 2]
```diff
@@ -142,9 +142,9 @@ export default class CollectionSearchBar extends mixins(KeyboardEventsMixin) {
   @Emit('update:listed')
   @Debounce(50)
   updateListed(value: string | boolean): boolean {
-    const v = String(value)
+    const v = String(value) === 'true'
```
Comment (roiLeo) [inline components/rmrk/Gallery/Search/SearchBarCollection.vue:145]: could you change const name to fix deepsource error?

File: components/rmrk/Gallery/CollectionItem.vue (hunk 1 above)
Comment (kkukelka) [reply components/rmrk/Gallery/CollectionItem.vue:311]: ahhh yes you are right

File: components/rmrk/Gallery/Search/SearchBarCollection.vue (hunk 2 above)
Comment (kkukelka) [reply components/rmrk/Gallery/Search/SearchBarCollection.vue]: done

File: components/rmrk/Gallery/Search/SearchBarCollection.vue [hunk 3]
```diff
@@ -142,17 +142,17 @@ export default class CollectionSearchBar extends mixins(KeyboardEventsMixin) {
   @Emit('update:listed')
   @Debounce(50)
   updateListed(value: string | boolean): boolean {
-    const v = String(value)
-    this.replaceUrl(v, 'listed')
-    return v === 'true'
+    const queryValue = String(value) === 'true'
+    this.replaceUrl(queryValue, 'listed')
+    return String(value) === 'true'
```
Comment (roiLeo) [inline components/rmrk/Gallery/Search/SearchBarCollection.vue:147]:
```suggestion
    return queryValue
```

**Review by roiLeo — APPROVED** [review]
✅ lgtm

**Comment by roiLeo** [comment]
pay 50

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 48.78 USD/KSM ~ 1.025 $KSM
🧗 [EqdyzrzVmeHwMdMwvPeCMnNdbuQDbD3YrjY93xq9Ln3jUGW](https://kusama.subscan.io/account/EqdyzrzVmeHwMdMwvPeCMnNdbuQDbD3YrjY93xq9Ln3jUGW)
🔗 [0xf6ee0d02ee619ea2c8b5e096f8d1b0bf5d0439c2f5c9d2cbf1f00072ec8729fd](https://kusama.subscan.io/extrinsic/0xf6ee0d02ee619ea2c8b5e096f8d1b0bf5d0439c2f5c9d2cbf1f00072ec8729fd)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**